cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java

```diff
@@ -182,10 +193,46 @@ public synchronized String predict(Bitmap image, Context context) {
             try {
                 return runModel(image);
             } catch (Error | Exception e) {
-                Log.i("ObjectDetect", "runModel exception, retry object detection", e);
+                Log.i("OCR", "runModel exception, retry object detection", e);
                 ssdOcrModel = new SSDOcrModel(context);
                 return runModel(image);
             }
+        } catch (Error | Exception e) {
+            Log.e("OCR", "unrecoverable exception on ObjectDetect", e);
+            hadUnrecoverableException = true;
+            return null;
+        }
+    }
+
+    public synchronized String predict(Bitmap image, Context context, boolean strict) {
```

Comment:
since `strict` defaults to `true`, can we replace the `predict(image, context)` method above this with a call to this method?

```
public synchronized String predict(Bitmap image, Context context) {
    return predict(image, context, true)
}
```